ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/rules/ChainMethodContinuation.kt

```diff
@@ -0,0 +1,195 @@
+import com.pinterest.ktlint.rule.engine.core.api.ElementType.SAFE_ACCESS
+import com.pinterest.ktlint.rule.engine.core.api.Rule
+import com.pinterest.ktlint.rule.engine.core.api.firstChildLeafOrSelf
+import com.pinterest.ktlint.rule.engine.core.api.hasNewLineInClosedRange
+import com.pinterest.ktlint.rule.engine.core.api.indent
+import com.pinterest.ktlint.rule.engine.core.api.isWhiteSpace
+import com.pinterest.ktlint.rule.engine.core.api.isWhiteSpaceWithNewline
+import com.pinterest.ktlint.rule.engine.core.api.lastChildLeafOrSelf
+import com.pinterest.ktlint.rule.engine.core.api.leavesIncludingSelf
+import com.pinterest.ktlint.rule.engine.core.api.nextCodeSibling
+import com.pinterest.ktlint.rule.engine.core.api.noNewLineInOpenRange
+import com.pinterest.ktlint.rule.engine.core.api.prevCodeSibling
+import com.pinterest.ktlint.rule.engine.core.api.prevLeaf
+import com.pinterest.ktlint.rule.engine.core.api.upsertWhitespaceBeforeMe
+import com.pinterest.ktlint.ruleset.standard.StandardRule
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+import org.jetbrains.kotlin.com.intellij.psi.tree.TokenSet
+
+/**
+ * When present, align chain operators('.' or '?.') at previous line with next method call
+ */
+public class ChainMethodContinuation :
+    StandardRule("chain-method-continuation"),
+    Rule.Experimental {
+    private val chainOperators = TokenSet.create(DOT, SAFE_ACCESS)
+
+    override fun beforeVisitChildNodes(
+        node: ASTNode,
+        autoCorrect: Boolean,
+        emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit,
+    ) {
+        node
+            .takeIf { it.elementType in chainOperators }
+            ?.let { visitChainOperator(node, autoCorrect, emit) }
+    }
+
+    private fun visitChainOperator(
+        chainOperator: ASTNode,
+        autoCorrect: Boolean,
+        emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit,
+    ) {
+        val rightExpression = chainOperator.nextCodeSibling()?.firstChildLeafOrSelf() ?: return
+        val previousEndBrace =
+            chainOperator.getPrevChainEndBrace() ?: return
+        val previousExpressionBeforeBrace = previousEndBrace.prevCodeSibling()?.lastChildLeafOrSelf() ?: return
+        val isPreviousChainElementMultiline =
+            previousEndBrace.elementType == RBRACE &&
+                hasNewLineInClosedRange(previousExpressionBeforeBrace, previousEndBrace)
+        if (hasNewLineInClosedRange(chainOperator, rightExpression)) {
+            /*
+                Detects code like below
+                bar {
+                    ...
+                }.
+                foo() // unexpected new line after .
+
+                or
+                bar(). // expected newline before .
+                foo()
+             */
+            fixMisalignedChainOperator(
+                emit,
+                chainOperator,
+                isPreviousChainElementMultiline,
```

Comment:
I have passed the required previously calculated value in the method let me know if you want this to be recalculated from `chainOperator` itself from some refactored extension method. To me this also reads fine

Reply:
yes, it is acceptable.